homeassistant/components/dyson_local/__init__.py

```diff
@@ -0,0 +1,117 @@
+"""Support for Dyson devices."""
+from __future__ import annotations
+
+from libdyson import MessageType, get_device
+from libdyson.dyson_device import DysonDevice
+from libdyson.exceptions import DysonException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.entity import DeviceInfo, Entity
+
+from .const import CONF_CREDENTIAL, CONF_DEVICE_TYPE, CONF_SERIAL, DATA_DEVICES, DOMAIN
+
+_PLATFORMS = ["binary_sensor", "sensor", "vacuum"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict) -> bool:
+    """Set up Dyson integration."""
+    hass.data[DOMAIN] = {
+        DATA_DEVICES: {},
+    }
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Dyson from a config entry."""
+    device = get_device(
+        entry.data[CONF_SERIAL],
+        entry.data[CONF_CREDENTIAL],
+        entry.data[CONF_DEVICE_TYPE],
+    )
+
+    try:
+        await hass.async_add_executor_job(device.connect, entry.data[CONF_HOST])
+    except DysonException as err:
+        raise ConfigEntryNotReady from err
+    hass.data[DOMAIN][DATA_DEVICES][entry.entry_id] = device
+    for component in _PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload Dyson local."""
+    device = hass.data[DOMAIN][DATA_DEVICES][entry.entry_id]
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, _PLATFORMS)
+    if unload_ok:
+        hass.data[DOMAIN][DATA_DEVICES].pop(entry.entry_id)
+        await hass.async_add_executor_job(device.disconnect)
+    return unload_ok
+
+
+class DysonEntity(Entity):
```

Comment:
Use entity class or instance attributes instead of properties if possible:
https://developers.home-assistant.io/docs/core/entity/#entity-class-or-instance-attributes